File: uStock-Development-Android/app/src/main/java/data_structures/Comment.kt
package data_structures

import data_structures.DateSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName
import java.util.Date

@Serializable
data class Comment(
    @SerialName("_id") val id: String,
    val author: String,
    val content: String,
    val post: String,
    @Serializable(with = DateSerializer::class)
    val timestamp: Date,
    val upvotes: List<String>,
    val downvotes: List<String>,
    val replies: List<String>
)

File: uStock-Development-Android/taskThree_models/Data Structures/Comment.swift
//
//  Comment.swift
//  uStockIOS-V2-AI
//
//  Created by Gal Cohavy on 6/7/23.
//


import Foundation

public struct Comment: Codable {
    let id: String
    let author: String
    let content: String
    let post: String
    let timestamp: Date
    let upvotes: [String]
    let downvotes: [String]
    let replies: [String]
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case author, content, post, timestamp, upvotes, downvotes, replies
    }
}
